packages/services/api/src/modules/preflight-script/providers/preflight-script.provider.ts

```diff
@@ -0,0 +1,215 @@
+import { Inject, Injectable, Scope } from 'graphql-modules';
+import { sql, type DatabasePool } from 'slonik';
+import { z } from 'zod';
+import { fromZodError } from 'zod-validation-error';
+import { getLocalLang, getTokenSync } from '@nodesecure/i18n';
+import * as jsxray from '@nodesecure/js-x-ray';
+import {
+  CreatePreflightScriptInput,
+  TargetSelectorInput,
+  UpdatePreflightScriptInput,
+} from '../../../__generated__/types';
+import { Session } from '../../auth/lib/authz';
+import { IdTranslator } from '../../shared/providers/id-translator';
+import { Logger } from '../../shared/providers/logger';
+import { PG_POOL_CONFIG } from '../../shared/providers/pg-pool';
+import { Storage } from '../../shared/providers/storage';
+
+const PreflightScriptModel = z.strictObject({
+  id: z.string(),
+  sourceCode: z.string().max(5_000),
+  targetId: z.string(),
+  createdByUserId: z.union([z.string(), z.null()]),
+  createdAt: z.string(),
+  updatedAt: z.string(),
+});
+
+const scanner = new jsxray.AstAnalyser();
+await getLocalLang();
+
+function validateSourceCode(code: string) {
```

Comment:
I think you can just make this as a [`.superRefine`](https://zod.dev/?id=superrefine) for the zod

Reply:
actually keep it how you have it that way we can return errors early and not have to do a full zod parsing